app/src/main/java/com/navapp/navigation/util/math/ClusteringUtil.java

```diff
@@ -0,0 +1,24 @@
+package com.navapp.navigation.util.math;
+
+import com.navapp.navigation.util.math.TSP.TSPSolver;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+
+public class ClusteringUtil {
```

Comment:
Util classes such as this are usually just shortcuts to good design. Make a proper class. Some of the parameters are obvious dependencies, while others are normal parameters.